### DISCUSSION TITLE
Needs

### DISCUSSION THREAD

**Discussion by mcguirepr89** [mcguirepr89/BirdNET-Pi#212] [Q&A]
- Someone to help Dockerize this - #211 
- Someone to help optimize package management (e.g., get this into a .deb package / `apt` repo)
- Better disclaimers regarding responsible use and users' onus for security, privacy, and respecting the license
- Better interfaces other than just the web (namely, adding arguments to most tools to support more dynamic input and output for the command line)
- Someone to help tinker with building an API would be nice

**Comment by saintaardvark** [comment]
Hi @mcguirepr89  -- I came across your project last week via HackerNews, and I'm intrigued.  My background is Linux system administration  and Python development, with a strong desire to learn more about ML.  I think I might be able to help out with some of the back-end tasks you list here -- packaging, CLI tools, and possibly helping out with an API.  Let me know the best way to get started!

**Reply by mcguirepr89** [reply]
That is great news!!

Where to start? Hmmmm :thinking:

I _think_ these items are in order by priority , but I may be wrong.

Item 1: Making Collaboration Easier
- I think it would be nice to have a Flask API with routes/endpoints to query and potentially update the database (This would obviously open up a TON of development around BirdNET-Pi for folks interested in pulling the data in with GET requests and to be able to start exploring a noded BirdNET-Pi network with a centralized database). I'm happy to handle any reverse_proxy needs to get everything served over SSL (haha, actually, `caddy` will just do it)
- Likewise, it would be nice to be able to `jsonify` the `analyze.py` output in addition to (or to supplant the) current `.csv` output and make it consumable 

Item 2: Making the installation smarter (using package management tools)
-  I know that 90% of the time, the installation works every time! (haha I couldn't resist that hilarity), meaning my concern with the current installation method is not its success rate, I simply know it can be done better. At its core, BirdNET-Pi is just a set of (mostly bash) scripts and system services with a configuration file, and then some PHP to show what's in the database and adjust the configuration. To me that screams `I'm A .deb package`. I would like the `curl` one liner to be adding the `.deb` repo to `apt-sources` and then `apt` handle dependencies (except for the part where it builds the python virtual environment and installs all those dependencies . . . you'll need to help me know what the best way to handle that is because I'm clueless). Anywho, I need to know more about this before I can do much about those wishes, though I thought I'd put them out there for you in case you want it.

Thank you again for your kind offer to contribute and please let me know how I can help

**Reply by saintaardvark** [reply]
I think I agree with the order of importance; my ability to help with Item 2 is definitely higher than Item 1.  I'll do some digging to remind myself how .debs get created (and figure out how one does this with Python packages), and will report back here.

**Reply by CaiusX** [reply]
Hi @saintaardvark - lets keep the momentum going on Patricks's couple of ideas, I'll start looking at Item 1, although I think the second bullet's already been largely done in the birdweather code section,

**Reply by saintaardvark** [reply]
Hi @mcguirepr89 -- Finally got some time to start working on this a bit.  I'm going to start up another discussion to ask some questions and discuss the process as I do (and learn!) more.

**Reply by saintaardvark** [reply]
Hi @mcguirepr89 -- I'm very sorry about this, but the amount of free time I have has changed, and I won't be able to work on this like I thought I would.  :cry: I wish you & this project the very best, and hope to be able to contribute in the future.
